### PR TITLE
Android Lollipop support (API 21)

### LINKED ISSUE #4: Android Lollipop support
Hi. I would like to ask why you guys dropped Android Lollipop support for this? I know that there have crashes when using WebView, and can be resolved by adding some on XML (based on my experience with version 1)

### PR DESCRIPTION
**Pull request by jcchikikomori**
Hi. I wanted to maintain the SDK for older apps such as [REDACTED]Delivery who has an older version of SDK
So here's my proposal PR

Changes:
- Convert `Activity` superclass into `AppCompatActivity`
- Rename `minSdkVersion` into 21
- Added toolbar for older devices who has no virtual navigation bar

Screenshots:
Running Android Lollipop 5.0 on Genymotion Cloud

Screen Shot 2020-08-26 at 11 20 36 PM
Screen Shot 2020-08-26 at 11 19 46 PM

### CODE REVIEW

**Comment by jcchikikomori** [comment]
This solves issue #4

**Comment by rjarce** [comment]
Hi @jcchikikomori , thanks for the PR, will take a look

File: sdk/src/main/java/com/paymaya/sdk/android/common/internal/screen/PayMayaPaymentActivity.kt [hunk 1]
```diff
@@ -68,6 +72,16 @@ internal abstract class PayMayaPaymentActivity<R : PayMayaRequest> :
         presenter.backButtonPressed()
     }
 
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                presenter.backButtonPressed()
+                super.onOptionsItemSelected(item)
```
Comment (paulocode) [inline sdk/src/main/java/com/paymaya/sdk/android/common/internal/screen/PayMayaPaymentActivity.kt]:
```suggestion
                true
```

File: sdk/src/main/java/com/paymaya/sdk/android/common/internal/screen/PayMayaPaymentActivity.kt [hunk 2]
```diff
@@ -77,6 +91,13 @@ internal abstract class PayMayaPaymentActivity<R : PayMayaRequest> :
         payMayaPaymentActivityWebView.settings.javaScriptEnabled = true
         payMayaPaymentActivityWebView.settings.allowFileAccess = true
         payMayaPaymentActivityWebView.webViewClient = WebViewClientImpl()
+
+        val toolbar = findViewById<Toolbar>(com.paymaya.sdk.android.R.id.toolbar)
+        setSupportActionBar(toolbar)
+        if (supportActionBar != null) {
+            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+            supportActionBar!!.title = ""
+        }
```
Comment (paulocode) [inline sdk/src/main/java/com/paymaya/sdk/android/common/internal/screen/PayMayaPaymentActivity.kt]:
```suggestion
        supportActionBar?.apply {
          setDisplayHomeAsUpEnabled(true)
          title = ""
        }
```

**Comment by paulocode** [comment]
Apologies for the delay @jcchikikomori. I added minor suggestions, but review is still ongoing.

**Comment by paulocode** [comment]
Hi @jcchikikomori, regarding the added toolbar, may we know which specific device this is intended for? We might not need it upstream as afaik most android devices have navigation bars.

**Comment by jcchikikomori** [comment]
> Hi @jcchikikomori, regarding the added toolbar, may we know which specific device this is intended for? We might not need it upstream as afaik most android devices have navigation bars.

I've seen McDelivery PH that uses PayMaya SDK v1 (based on their UI), and it has a toolbar. So it might confuse the users like what is happening with that inconsistency.

I made a mistake that I include that on this PR instead of separating it.

Edit: I would suggest that make the toolbar feature as optional, and the default value is hidden

**Comment by paulocode** [comment]
Hi @jcchikikomori, 

1. Kindly remove toolbar changes for now. This will be discussed in a separate issue/PR. 

2. Regarding `uiMode` (#4), I am unable to replicate this crash on v2 SDK and 21 API level device. I think this crash is caused by an outdated dependency on v1 SDK and no longer applies to v2. Please share crash logs (if any).

3. Upon testing, it seems that just lowering the minSdkVersion to 21 is enough to support this API level. Do you have other concerns if we do just this? (EDIT: for further testing)

Thanks for your inputs.

**Comment by jcchikikomori** [comment]
Hi @paulocode. I just updated my PR by removing the toolbar. You may disregard the past screenshots I have. You can replace it if you want.

Thank you for your time reviewing my PR.

File: sdk/src/main/java/com/paymaya/sdk/android/common/internal/screen/PayMayaPaymentActivity.kt [hunk 3]
```diff
@@ -77,6 +81,15 @@ internal abstract class PayMayaPaymentActivity<R : PayMayaRequest> :
         payMayaPaymentActivityWebView.settings.javaScriptEnabled = true
         payMayaPaymentActivityWebView.settings.allowFileAccess = true
         payMayaPaymentActivityWebView.webViewClient = WebViewClientImpl()
+
+        /* TODO: Make a option to use toolbar (either through config, etc.)
```
Comment (paulocode) [inline sdk/src/main/java/com/paymaya/sdk/android/common/internal/screen/PayMayaPaymentActivity.kt]: Please remove TODO & commented code, then add an issue tagged as enhancement. We will track it there. Thanks.

Comment (paulocode) [inline sdk/src/main/java/com/paymaya/sdk/android/common/internal/screen/PayMayaPaymentActivity.kt:85]: Please remove TODO & commented code. We will track this via issue/PR instead.

File: sdk/src/main/res/layout/activity_paymaya_payment.xml [hunk 4]
```diff
@@ -1,49 +1,55 @@
 <?xml version="1.0" encoding="utf-8"?>
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (paulocode) [inline sdk/src/main/res/layout/activity_paymaya_payment.xml]: Please also revert changes to this file. PR should be okay to merge afterwards. Thanks.

**Pull request merged**